File: ObfuscateConnectionData/server/aspnet/RevealSdk.Server/Reveal/Encoder.cs
using Reveal.Sdk;

namespace RevealSdk.Server.Reveal
{
    public class Encoder : IRVObjectEncoder
    {
        public Task<RVDashboardDataSource> Encode(IRVUserContext userContext, RVDashboardDataSource dataSource)
        {
            if (dataSource is RVSqlServerDataSource sqlDataSource)
                EncodeDataSource(sqlDataSource);
            
            return Task.FromResult(dataSource);
        }

        public Task<RVDataSourceItem> Encode(IRVUserContext userContext, RVDataSourceItem dataSourceItem)
        {
            if (dataSourceItem is RVSqlServerDataSourceItem sqlDataSourceItem)
            {
                var sqlDataSource = (RVSqlServerDataSource)sqlDataSourceItem.DataSource;
                EncodeDataSource(sqlDataSource);

                sqlDataSourceItem.Database = "DB";
                sqlDataSourceItem.Schema = "SCHEMA";
            }
            return Task.FromResult(dataSourceItem);
        }

        private void EncodeDataSource(RVSqlServerDataSource sqlDataSource)
        {
            sqlDataSource.Host = "HOST";
            sqlDataSource.Database = "DB";
            sqlDataSource.Schema = "SCHEMA";
        }
    }
}

File: ObfuscateConnectionData/server/spring-boot-jersey/src/main/java/com/server/reveal/Encoder.java
package com.server.reveal;

import com.infragistics.reveal.sdk.api.IRVObjectEncoder;
import com.infragistics.reveal.sdk.api.IRVUserContext;
import com.infragistics.reveal.sdk.api.model.RVDashboardDataSource;
import com.infragistics.reveal.sdk.api.model.RVDataSourceItem;
import com.infragistics.reveal.sdk.api.model.RVSqlServerDataSource;
import com.infragistics.reveal.sdk.api.model.RVSqlServerDataSourceItem;

public class Encoder implements IRVObjectEncoder {

    @Override
    public RVDashboardDataSource encode(IRVUserContext userContext, RVDashboardDataSource dataSource) {
        
        if (dataSource instanceof RVSqlServerDataSource sqlDataSource) {
            EncodeDataSource(sqlDataSource);
        }
        return dataSource;
    }

    @Override
    public RVDataSourceItem encode(IRVUserContext userContext, RVDataSourceItem dataSourceItem) {
        if (dataSourceItem instanceof RVSqlServerDataSourceItem sqlDataSourceItem)
        {
            var sqlDataSource = (RVSqlServerDataSource)sqlDataSourceItem.getDataSource();
            EncodeDataSource(sqlDataSource);

            sqlDataSourceItem.setDatabase("DB");
            sqlDataSourceItem.setSchema("SCHEMA");
        }
        return dataSourceItem;
    }

    private void EncodeDataSource(RVSqlServerDataSource sqlDataSource)
    {
        sqlDataSource.setHost("HOST");
        sqlDataSource.setDatabase("DB");
        sqlDataSource.setSchema("SCHEMA");
    }
}
